final external = '[shimmed at compile time]';

@external
class Exception {
    final message, stack;

    constructor(message) {}
}

class Iterator {
    let current, i, items;

    constructor(items) {
        this.items = items;
    }

    fn moveNext() {
        if (this.i < (this.items.length - 1)) {
            this.i += 1; // Todo: Unary this :)
            this.current = (this.items[this.i]);
            ret true;
        } else {
            ret false;
        }
    }

    fn peek() {
        if ((this.i + 1) < this.items.length) {
            ret this.items[this.i + 1];
        }

        ret null;
    }
}

@external
fn print(msg) {}

@external
final process;